specs/flyout/flyout.md

```diff
@@ -0,0 +1,179 @@
+
+
+### Prior Art/Examples
+
+#### Popover from Salesforce
+- https://www.lightningdesignsystem.com/components/popovers/
+
+![Popover](./images/salesforce-popover.png)
+#### Popover from Evergreen
+- triggered by a button
+- https://evergreen.segment.com/components/popover/
+
+![Popover](./images/evergreen-popover.png)
+
+#### Microsoft Windows Developer Flyout, Text Only
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+#### Microsoft Windows Developer Flyout, Actionable
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+#### Inline Dialog from ATLAssian
+- https://atlassian.design/components/inline-dialog/examples
+
+![Popover](./images/atlassian-popover.png)
+
+#### Callout with actions from Fluent UI
+- https://developer.microsoft.com/en-us/fluentui#/controls/web/callout
+![Callout](./images/fluent-callout.png)
+
+
+---
+
+### API
+*Component Name*
+- `fast-popover`
+
+*Attributes*
+- `visible` - boolean, whether or not the Popover is visible, defaults to undefined, as this is controlled by the author
+- `trap-focus` - boolean, whether or not to keep focus contained inside the popover, defaults to true
+- `target` - string, the html id of the HTMLElement that the Popover is attached to, (viewport for anchored region defaults to the parent of the anchor) and triggered by
+- `focus-target-on-close` - boolean, whether or not to focus back on to the _target element_ when the popover is closed, defaults to true
+- `position` - enum, where the tooltip should appear relative to its target, uses Anchored Region logic=
+- `delay` - number, the time delay before the popover is shown once triggered
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+    ** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Properties*
+- `targetElement` - the reference to the HTMLElement used as an target
+- `viewportElement` - the reference to the HTMLElement used as the viewport, required ot setup anchored region properly
+- `popoverVisible` - boolean, whether or not the Popover is visible
+
+*Slots*
+- default for content
+
+*Events*
+- `dismiss` - event fired when the Popover is dismissed
+
+*Functions*
+- `createPopoverTemplate(string: prefix)` - generates a `ViewTemplate` for the popover based on the given provided prefix string. This is required as popover uses an `anchored-region` internally and the create function generates a template using the appropriate `anchored-region` tag (ie "fast-anchored-region", "fluent-anchored-region"). Note that the appropriate `anchored-region` component must also be declared.
```

Comment:
`createTemplate` would make it easier to rename this component if we decide to go with something other than `popover`.

Reply:
We currently have 2 other create***Template functions that are public and exported so I think we need to have Popover in there for clarification.